web/src/components/navbar/NavBarPopupMenu.tsx

```diff
@@ -0,0 +1,188 @@
+        handleClose();
+      },
+      icon: <ButtonIcon svgFilename="logout" sizePx="25px" />,
+      itemText: Config.navigationDefaults.logoutButton,
+      key: "logoutMenuItem",
+    });
+  };
+
+  const myNjMenuItem = (): ReactElement => {
+    return NavMenuItem({
+      onClick: (): void => {
+        analytics.event.account_menu_myNJ_account.click.go_to_myNJ_home();
+        window.open(process.env.MYNJ_PROFILE_LINK || "", "_ blank");
+        handleClose();
+      },
+      icon: <ButtonIcon svgFilename="profile" sizePx="25px" />,
+      itemText: Config.navigationDefaults.myNJAccountText,
+      key: "myNjMenuItem",
+    });
+  };
+
+  const registerMenuItem = (): ReactElement => {
+    return NavMenuItem({
+      onClick: (): void => {
+        analytics.event.guest_menu.click.go_to_myNJ_registration();
+        onSelfRegister(router, updateQueue, setRegistrationAlertStatus);
+      },
+      icon: <ButtonIcon svgFilename="profile" sizePx="25px" />,
+      itemText: Config.navigationDefaults.navBarGuestRegistrationText,
+      key: "registerMenuItem",
+    });
+  };
+
+  const getStartedMenuItem = (): ReactElement => {
+    return NavMenuItem({
+      onClick: (): void => {
+        router.push(ROUTES.onboarding);
+      },
+      itemText: Config.navigationDefaults.registerButton,
+      key: "getStartedMnueItem",
+    });
+  };
+
+  const profileMenuItem = (): ReactElement[] => {
+    return [
+      <hr className="margin-0 hr-2px" key={"profile-break-1"} />,
+
+      NavMenuItem({
+        onClick: (): void => {
+          router.push(ROUTES.dashboard);
+        },
+        selected: isDashboardSelected,
+        justFocused: !isProfileSelected && !isDashboardSelected,
+        icon: <ButtonIcon svgFilename="business-green" sizePx="35px" />,
+        itemText: navBarBusinessTitle,
+        key: "dashboardMenuItem",
+      }),
+
+      NavMenuItem({
+        onClick: (): void => {
+          analytics.event.account_menu_my_profile.click.go_to_profile_screen();
+          router.push(ROUTES.profile);
+        },
+        selected: isProfileSelected,
+        itemText: Config.navigationDefaults.profileLinkText,
+        key: "profileMenuItem",
+        dataTestid: "profile-link",
+      }),
+
+      <hr className="margin-0 hr-2px" key={"profile-break-2"} />,
+    ];
+  };
+
+  const renderMenu = (): ReactElement[] | undefined | Array<ReactElement | ReactElement[]> => {
+    if (menuConfiguration === "login") {
+      return [loginMenuItem()];
+    }
+    if (menuConfiguration === "profile") {
+      return [profileMenuItem()];
+    }
+    if (menuConfiguration === "profile-myNJ-logout") {
+      return [profileMenuItem(), myNjMenuItem(), logoutMenuItem()];
+    }
+    if (menuConfiguration === "profile-register-login") {
+      return [profileMenuItem(), registerMenuItem(), loginMenuItem()];
+    }
+    if (menuConfiguration === "login-getstarted") {
+      return [getStartedMenuItem(), loginMenuItem()];
+    }
```

Comment:
This is enough logic that seems like it could be unit-tested for this component independently of the integration-tests that are in NavBar.test

Reply:
Sure